FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpDecryptAndOrVerify.kt

```diff
@@ -85,9 +91,23 @@ object PgpDecryptAndOrVerify {
             content = destOutputStream
           )
         } catch (e: Exception) {
-          return DecryptionResult.withError(
-            processDecryptionException(e)
-          )
+          if (retryCount > 0 && srcStream.markSupported()) {
+            srcStream.reset()
+            return decryptAndOrVerifyWithResult(
+              srcInputStream,
+              publicKeys,
+              secretKeys,
+              protector,
+              passphrase,
+              ignoreMdcErrors,
+              retryCount - 1
```

Comment:
`retryCount` initial count is 0 and we pass `-1` value? I think we should pass `retryCount + 1` value?

Reply:
Sorry?
`retryCount` initial value is 0. But you pass `retryCount` as `retryCount - 1` in above line.
Therefore I think `if (retryCount > 0 && srcStream.markSupported()) {` condition would never satisfy.

Reply:
> retryCount initial value is 0.

Yup. by default `retryCount = 0`. I don't want to use the retry strategy by default. So by default, we will try to decrypt a source only once(as we did it before).

Reply:
You are right.
I got confused because I didn't see `retryCount` passed as `positive integers`.
But now I realized this one passes positive retryCount value.
Everything looks good 👍 
Sorry for confusion

https://github.com/FlowCrypt/flowcrypt-android/blob/98477f1315e9094be812032e70200c6d19585d88/FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpMsg.kt#L873

Reply:
@DenBond7, I also got confused because of `retryCount` variable name.
Previously, I used similar `retry logic` by passing `retryCount + 1` and then comparing it to `MAX_RETRY_COUNT`. 

I think it would be better to use a more descriptive variable name, such as `remainingRetryCount`, `retryAttemptsLeft`, or `retryAttemptsRemaining`, to make it clearer that this variable represents the number of retry attempts remaining.

Reply:
What about `additionalRetryCount`? We have a base behavior and can have additional attempts for decryption.